### ISSUE TITLE
[Bug]: PP=2 + GlmMoeDsa: inductor compile combined with CUDA-graph capture produces garbage output; either alone is clean (v0.24 & v0.26)

### ISSUE THREAD

**Issue by divyvasal** [vllm-project/vllm#49844]
### Your current environment

- vLLM: reproduced on both **v0.24.0** and **v0.26.0** (official `vllm/vllm-openai` images)
- Model: GLM-5.2 (`GlmMoeDsaForCausalLM`, 744B MoE, 78 layers, 256 routed experts), NVFP4 compressed-tensors checkpoint (FP4 routed experts; attention / shared experts / dense layers BF16)
- Hardware: 2 × AWS p6-b200.48xlarge (8× B200 each, same AZ, cluster placement group)
- Parallelism: `--pipeline-parallel-size 2 --tensor-parallel-size 8 --enable-expert-parallel`, `--distributed-executor-backend ray` (PP stage 0 = node 0, stage 1 = node 1; verified via worker placement)
- NCCL 2.28.9; cross-node transport: NCCL Socket during bisection (EFA later — doesn't change the result pattern)
- `--block-size 64` required to boot at all (see #48286), `--kv-cache-dtype fp8` and default both tested

### 🐛 Describe the bug

Under PP=2, the engine boots and serves, but **every completion is garbage** — deterministic all-`!` output at temperature 0:

```json
"content": "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
```

(classic NaN-logits → argmax-stuck signature). The same checkpoint on the same image and hardware produces correct output single-node.

Bisection matrix (same checkpoint, same boxes, temperature 0 smoke each time):

| Config | Output |
|---|---|
| PP2×TP8×EP8, defaults (inductor + piecewise cudagraphs), v0.24 | garbage |
| PP2, `--kv-cache-dtype fp8` vs default | garbage both (not KV dtype) |
| **Single-node TP8, graphed, v0.24** | **clean** |
| **PP2 `--enforce-eager`, v0.24** | **clean** |
| PP2 graphed after wiping `~/.cache/vllm` on all nodes, v0.24 | garbage (not stale compile cache) |
| PP2 graphed, v0.26 | garbage (persists across versions) |
| **PP2, inductor ON + `--compilation-config '{"cudagraph_mode":"NONE"}'`, v0.26** | **clean** |
| **PP2, inductor OFF + `'{"mode":"NONE","cudagraph_mode":"FULL"}'`, v0.26** | **clean** |
| PP2, inductor ON + `'{"cudagraph_mode":"FULL"}'`, v0.26 | garbage |

So the corruption requires **inductor compilation × CUDA-graph capture together, under PP** — either alone is fine, and any capture mode (piecewise or FULL) combined with inductor is broken. Single-node TP8 with the identical default compile+graph stack is clean, so it's PP-specific.

### Workarounds

Either of:
- `--compilation-config '{"cudagraph_mode":"NONE"}'` (keep inductor, lose graphs), or
- `--compilation-config '{"mode":"NONE","cudagraph_mode":"FULL"}'` (keep graphs, lose inductor — what we're running).

Happy to provide full worker logs, the exact serve command, or run candidate fixes on this setup while we still have the 2-node rig up.
